File: frontend/src/types/report.ts
export interface BookingsPerFlightReport {
  flight_number: string;
  departure: string;
  arrival: string;
  booking_count: number;
  total_revenue: number;
}

export interface PopularRouteReport {
  origin: string;
  destination: string;
  origin_code: string;
  destination_code: string;
  origin_city: string;
  destination_city: string;
  total_bookings: number;
  booking_count: number;
}

export interface PeakTimeReport {
  hour: number;
  total_bookings: number;
  booking_count: number;
}

File: backend/app/schemas/report.py
from pydantic import BaseModel


class BookingsPerFlightReport(BaseModel):
    flight_id: int
    flight_number: str
    departure: str
    arrival: str
    booking_count: int


class PopularRouteReport(BaseModel):
    origin_code: str
    origin_city: str
    destination_code: str
    destination_city: str
    booking_count: int


class PeakTimeReport(BaseModel):
    hour: int
    booking_count: int
